server-timing/class-perflab-server-timing.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Server-Timing API: Perflab_Server_Timing class
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Class controlling the Server-Timing header.
+ *
+ * @since n.e.x.t
+ */
+class Perflab_Server_Timing {
+
+	/**
+	 * Map of registered metric slugs and their metric instances.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $registered_metrics = array();
+
+	/**
+	 * Map of registered metric slugs and their registered data.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $registered_metrics_data = array();
+
+	/**
+	 * Registers a metric to calculate for the Server-Timing header.
+	 *
+	 * This method must be called before the {@see 'perflab_server_timing_send_header'} hook.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $metric_slug The metric slug.
+	 * @param array  $args        {
+	 *     Arguments for the metric.
+	 *
+	 *     @type callable $measure_callback The callback that initiates calculating the metric value. It will receive
+	 *                                      the Perflab_Server_Timing_Metric instance as a parameter, in order to set
+	 *                                      the value when it has been calculated. Metric values must be provided in
+	 *                                      milliseconds.
+	 *     @type string   $access_cap       Capability required to view the metric. If this is a public metric, this
```

Comment:
should there be a default for this value if not passed?

Reply:
@adamsilverstein See my response to a similar question in https://github.com/WordPress/performance/pull/553#discussion_r1048626240, I intentionally did not want to add a default for that reason. Sometimes defaults can be rather detrimental (a good example is the `autoload = yes` default in core - would have been better if there was none, requiring developers to think about it.